### PR TITLE
Do not throw deprecation on validation for composer-runtime-api

### LINKED ISSUE #8841: Validate fail with composer-runtime-api
My `composer.json`:

```json
{
    "name": "test/test",
    "description": "test",
    "require": {
        "composer-runtime-api": "^2.0"
    }
}
```

Output of `composer diagnose`:

```
$ composer diagnose
Deprecation warning: require.composer-runtime-api is invalid, it should have a vendor name, a forward slash, and a package name. The vendor and package name can be words separated by -, . or _. The complete name should match "[a-z0-9]([_.-]?[a-z0-9]+)*/[a-z0-9]([_.-]?[a-z0-9]+)*". Make sure you fix this as Composer 2.0 will error.
Checking composer.json: WARNING
No license specified, it is recommended to do so. For closed-source software you may use "proprietary" as license.
Deprecation warning: require.composer-runtime-api is invalid, it should have a vendor name, a forward slash, and a package name. The vendor and package name can be words separated by -, . or _. The complete name should match "[a-z0-9]([_.-]?[a-z0-9]+)*/[a-z0-9]([_.-]?[a-z0-9]+)*". Make sure you fix this as Composer 2.0 will error.
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: 1.10.5
PHP version: 7.4.3
PHP binary path: /usr/bin/php7.4
OpenSSL version: OpenSSL 1.1.1d  10 Sep 2019
```

When I run this command:

```
composer validate
```

I get the following output:

```
Deprecation warning: require.composer-runtime-api is invalid, it should have a vendor name, a forward slash, and a package name. The vendor and package name can be words separated by -, . or _. The complete name should match "[a-z0-9]([_.-]?[a-z0-9]+)*/[a-z0-9]([_.-]?[a-z0-9]+)*". Make sure you fix this as Composer 2.0 will error.
Deprecation warning: require.composer-runtime-api is invalid, it should have a vendor name, a forward slash, and a package name. The vendor and package name can be words separated by -, . or _. The complete name should match "[a-z0-9]([_.-]?[a-z0-9]+)*/[a-z0-9]([_.-]?[a-z0-9]+)*". Make sure you fix this as Composer 2.0 will error.
./composer.json is valid, but with a few warnings
See https://getcomposer.org/doc/04-schema.md for details on the schema
The lock file is not up to date with the latest changes in composer.json, it is recommended that you run `composer update` or `composer update <package name>`.
No license specified, it is recommended to do so. For closed-source software you may use "proprietary" as license.
Deprecation warning: require.composer-runtime-api is invalid, it should have a vendor name, a forward slash, and a package name. The vendor and package name can be words separated by -, . or _. The complete name should match "[a-z0-9]([_.-]?[a-z0-9]+)*/[a-z0-9]([_.-]?[a-z0-9]+)*". Make sure you fix this as Composer 2.0 will error.
```

And I expected this to happen:
 * no deprecatons errors on the `composer-runtime-api` requirement

### PR DESCRIPTION
**Pull request by Jean85**
Fixes #8841

### CODE REVIEW

**Comment by stof** [comment]
Shouldn't this be handled the same than t for `composer-plugin-api` ?

**Comment by Jean85** [comment]
@stof the only reference of this kind that I can find is this one: https://github.com/composer/composer/blob/0ab48a1773e55ee1706d43e739fa5719811a87c8/src/Composer/Plugin/PluginManager.php#L124-L127

I cannot find any special handling for that, but the validation doesn't fail...

File: src/Composer/Package/Loader/ValidatingArrayLoader.php [hunk 1]
```diff
@@ -370,6 +370,10 @@ public function getErrors()
 
     public static function hasPackageNamingError($name, $isLink = false)
     {
+        if ($name === 'composer-runtime-api') {
+            return;
+        }
+
```
Comment (Seldaek) [inline src/Composer/Package/Loader/ValidatingArrayLoader.php:376]: This is not needed. PLATFORM_PACKAGE_REGEX includes/matches `composer-runtime-api`

Comment (Seldaek) [reply src/Composer/Package/Loader/ValidatingArrayLoader.php:376]: By this I mean.. if it is needed then something is wrong there, but such a workaround should not be necessary.

Comment (Seldaek) [reply src/Composer/Package/Loader/ValidatingArrayLoader.php:376]: Ah I see now looking at your issue, you were using 1.10.5, which is not aware of composer-runtime-api yet, so yes that one will output a warning. So this PR should target 1.10 branch rather, updating the regex to match that of master..

Comment (Jean85) [reply src/Composer/Package/Loader/ValidatingArrayLoader.php:376]: Roger! I thought master was still 1.x, I missed that I needed to change that. I will fix this ASAP.

**Comment by Jean85** [comment]
CI Failure seems unrelated, requesting another review.

**Pull request merged**